demo/src/client/QueryModelPage/ExampleGridPanel.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { PureComponent } from 'react';
+import { GridPanel, InjectedQueryModels, withQueryModels } from '@labkey/components';
+
+// First define the properties of the component you're going to wrap with withQueryModels.
+interface OwnProps {
+    title: string;
+    asPanel: boolean;
+}
+
+// Here we create a type that includes InjectedQueryModels because
+// we're wrapping the component with withQueryModels which will inject
+// queryModels and actions objects.
+type Props = OwnProps & InjectedQueryModels;
+
+// Here we use the name ExampleGridPanelImpl, users will not use this
+// component directly, only the wrapped version below which we expose
+// to users as ExampleGridPanel.
+class ExampleGridPanelImpl extends PureComponent<Props> {
+    onRefreshGrid = () => {
+        const { queryModels, actions } = this.props;
+        const { containersModel } = queryModels;
+
+        actions.loadModel(containersModel.id);
+    };
+
+    // This is an example of a custom button bar element in your GridPanel that can interact with the QueryModel.
+    renderGridButtons = () => {
+        return (
+            <button className={'labkey-button'} onClick={this.onRefreshGrid}>
+                Refresh Grid
+            </button>
+        )
+    };
+
+    render() {
+        const { queryModels, actions, title, asPanel } = this.props;
```

Comment:
Is it worth a comment here that queryModels and actions come from InjectedQueryModels?

Reply:
added